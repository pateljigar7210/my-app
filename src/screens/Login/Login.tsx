
import React, { useEffect, useState } from 'react';
import { View, Text, TextInput, StyleSheet, SafeAreaView, Keyboard, TouchableOpacity } from 'react-native';
import { ListOfAstronod, NearEarthObjectsEntity } from './types/AstronodListType'
import { Astronod } from "./types/Astronod";
import { NavigationProp } from '@react-navigation/native';
import ProgressDiloag from '../../components/ProgressDiloag';
import AlertDiload from '../../components/AlertDiload';

import 'react-native-gesture-handler';
interface LoginScreenProps {
    navigation: NavigationProp<any, any>,
}

const Login = ({ navigation }: LoginScreenProps) => {

    const [astronodId, setAstronodId] = useState<string>('')
    const [isLoding, setIsLoding] = useState<boolean>(false)
    const [message, setMessage] = useState<string>('')
    const [showMessage, setShowMessage] = useState<boolean>(false)


    useEffect(() => {
        return setIsLoding(false)
    }, [])
    const getListApi = () => {
        Keyboard.dismiss()
        setIsLoding(true)
        fetch('https://api.nasa.gov/neo/rest/v1/neo/browse?api_key=QMbqJOKAJdJepZ6XKhV23xl6atyA3N2Y2b8wZGuk')
            .then((response) => response.json())
            .then((json) => {
                const data = json as ListOfAstronod;
                if (data.near_earth_objects?.length != 0) {
                    var length = data.near_earth_objects ? data.near_earth_objects.length : 0;
                    const randomPosition = Math.floor(Math.random() * length)
                    const randomObj = data?.near_earth_objects ? data.near_earth_objects[randomPosition] : null
                    if (randomObj) {
                        const ojects = randomObj as NearEarthObjectsEntity;
                        setAstronodId(ojects.id)
                        getSingleAstronod(ojects.id)
                    }
                }
            }).catch((e) => {
                setIsLoding(false)
            }).finally(() => {
                setIsLoding(false)
            })

    }
    const getSingleAstronod = (astronodId: string) => {
        Keyboard.dismiss()
        setIsLoding(true)
        fetch(`https://api.nasa.gov/neo/rest/v1/neo/${astronodId}?api_key=QMbqJOKAJdJepZ6XKhV23xl6atyA3N2Y2b8wZGuk`)
            .then((response) => response.json())
            .then((json) => {
                setIsLoding(false)

                navigation.navigate('Home', {
                    data: json
                })


            }).catch((e) => {
                setIsLoding(false)
                setMessage('Wrong Asteroid ID')
                setShowMessage(true)
            }).finally(() => {
                setIsLoding(false)
            })

    }


    return (
        <SafeAreaView>

            <View style={styles.loginContainer}>
                <TextInput
                    style={styles.input}
                    value={astronodId}
                    onChangeText={setAstronodId}
                    placeholder='Enter Asteroid ID'
                />

                <View style={styles.buttonContainer}>

                    <TouchableOpacity
                        style={astronodId == '' ? styles.btnSubmitBlur : styles.btnSubmit}
                        onPress={() => getSingleAstronod(astronodId)}
                        disabled={astronodId == '' ? true : false}
                    >
                        <Text style={styles.btnSubmitText} >Submit</Text>
                    </TouchableOpacity>
                    <TouchableOpacity
                        onPress={() => getListApi()}
                        style={styles.btnSubmit}>
                        <Text style={styles.btnSubmitText} >Random Asteroid</Text>
                    </TouchableOpacity>
                </View>


            </View>
            <ProgressDiloag
                isViseble={isLoding}
            />
            <AlertDiload
                message={message}
                okPress={() => setShowMessage(false)}
                isViseble={showMessage}
            />
        </SafeAreaView>
    );
}

export default Login;


const styles = StyleSheet.create({
    loginContainer: {
        margin: 20,
        backgroundColor: '#FFF',
        padding: 20,
        borderRadius: 20,

    },
    input: {
        height: 46,
        margin: 12,
        borderWidth: 1,
        padding: 10,
        borderRadius: 10,
    },
    buttonContainer: {
        flexDirection: 'row'
    },
    btnSubmit: {
        backgroundColor: 'black',
        padding: 14,
        alignItems: 'center',
        margin: 12,
        height: 46,
        borderRadius: 10,
        elevation: 4,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.8,
        shadowRadius: 1,
    },

    btnSubmitBlur: {
        backgroundColor: 'gray',
        padding: 14,
        alignItems: 'center',
        margin: 12,
        height: 46,
        borderRadius: 10,
        elevation: 4,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.8,
        shadowRadius: 1,
    },
    btnSubmitText: {
        color: '#FFF',

    },
});
