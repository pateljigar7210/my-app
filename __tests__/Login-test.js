import React from 'react';
import renderer from 'react-test-renderer';

import 'react-native-gesture-handler';
import  Login from '../src/screens/Login/Login';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));
jest.mock('react-native-gesture-handler', () => {});

it('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});