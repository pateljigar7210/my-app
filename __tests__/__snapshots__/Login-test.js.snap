// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 20,
        "margin": 20,
        "padding": 20,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter Asteroid ID"
      rejectResponderTermination={true}
      style={
        Object {
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 46,
          "margin": 12,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "gray",
            "borderRadius": 10,
            "elevation": 4,
            "height": 46,
            "margin": 12,
            "opacity": 1,
            "padding": 14,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
            }
          }
        >
          Submit
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 10,
            "elevation": 4,
            "height": 46,
            "margin": 12,
            "opacity": 1,
            "padding": 14,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
            }
          }
        >
          Random Asteroid
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
        <Text
          style={
            Object {
              "marginTop": 15,
              "textAlign": "center",
            }
          }
        >
          Please Wait...
        </Text>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
        >
          
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 2,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#2196F3",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
                "width": 60,
              }
            }
          >
             Ok 
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</RCTSafeAreaView>
`;
